src/utils.rs

```diff
@@ -113,6 +113,16 @@ pub fn subdomain_match(subdomain: &str, authority: &str) -> bool {
     authority.ends_with(subdomain.replace('*', "").as_str())
 }
 
+pub fn path_match(path_pattern: &str, request_path: &str) -> bool {
+    if path_pattern.ends_with('*') {
+        let mut cp = path_pattern.to_string();
+        cp.pop();
```

Comment:
The `to_string()` here isn't really necessary, you want to get rid of the last char, you can keep it as a slice and just narrow this slice: `&path_pattern[..path_pattern.len() - 1]` which returns a slice…

Reply:
… which saves one heap allocation, this points to the same `str` wherever it sits…